models/result.go

```diff
@@ -58,6 +58,26 @@ func GetResultById(id int64) (*Result, error) {
 	return result, nil
 }
 
+// GetResultByIdWithRelations retrieves Result by Id with assigned relations. Returns error if Id doesn't exist
+func GetResultByIdWithRelations(id int64) (*Result, error) {
+	result, err := GetResultById(id)
+	if err != nil {
+		return result, err
+	}
+
+	result.Links, err = GetLinksByResultId(result.Id)
+	if err != nil {
+		return result, err
+	}
+
+	result.AdLinks, err = GetAdLinksByResultId(result.Id)
+	if err != nil {
+		return result, err
+	}
+
+	return result, nil
```

Comment:
Should not be the both value `nil` as the error is present at this point?
```suggestion
	return nil, nil
```

Reply:
@junan What do you mean? I think the error is not present at this line and that means everything went fine so we suppose to return the result, no?
or do you mean [line 65](https://github.com/carryall/go-google-scraper-challenge/pull/72/files#diff-e5575e255237767a833fc7e25a58186e9e03157369c630efd4f479e07f5029caR65)?

Reply:
>or do you mean line 65? if yes I add an update on 61f4857 😄
Yes 👍